pkg/telemetryquerylanguage/tqlconfig/config.go

```diff
@@ -0,0 +1,27 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tqlconfig // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tqlconfig"
+
+// Config configures the TQL queries to execute against traces, metrics, and logs.
+type Config struct {
+	Traces  SignalConfig `mapstructure:"traces"`
+	Metrics SignalConfig `mapstructure:"metrics"`
+	Logs    SignalConfig `mapstructure:"logs"`
+}
+
+// SignalConfig configures TQL queries to execute.
+type SignalConfig struct {
```

Comment:
Do we ever want to have a different config between signals?

Reply:
The current config allows configuring different queries per signal, but thats it so far.  Haven't come across a use-case yet for the config beside defining the queries themselves.

Reply:
Are you thinking instead of a `SignalConfig` we'd have `TracesConfig`, `MetricsConfig`, and `LogsConfig`?  At the moment all 3 would need the `Queries` property and would be duplicates of one another.  Have anything in mind that we'd add to one but not the other?

Reply:
Should we consider having each of Traces, Metrics, and Logs be an interface instead of a struct? That would be something like:
```go
type Queryer interface {
    GetQueries() []string
}
```
Or does that screw up something relating to loading the configurations?

Reply:
I believe they need to be structs to interact correctly with `component.ProcessorFactory`